sdk/core/Azure.Core/samples/Serialization.md

```diff
@@ -1,34 +1,47 @@
-# Azure.Core public serialization samples
+# Azure.Core Public Serialization Guide
 
-## Using explicit cast
+The latest updates to the `Azure.Core` library have simplified serializing and deserializing of all public Azure models. With the addiion of the `ModelSerializer` class, we provide a public interface that exposes the currently internal serialization code so customers can access it without needing to use reflection or write their own translation. The recommended serialization approach is to use the static `ModelSerializer` class. The `ModelSerializer` class also has the `ModelSerializerOptions` which allows the user to set the serialization `ModelSerializerFormat` type (`Json` or `Wire`) and provide custom Serializer types for specific models.
+
+## Key Concepts
+
+- [Using the ModelSerializer](#using-the-modelserializer)
+- [Using explicit cast](#using-explicit-cast)
+- [Using the ModelJsonConverter](#using-the-modeljsonconverter)
+- [Envelope BYOM Case](#envelope-byom-case)
+- [Using ModelSerializer with generic IModelSerializable](#using-modelserializer-with-generic-imodelserializable)
 
-When using protocol methods for advanced handling of RequestContext it is still possible to use the strongly typed models.
-There is an explicit cast operator that can be used to convert the protocol Response to the strongly typed model.
-There is also an explicit cast operator that can be used to convert the strongly typed model to the protocol RequestContent.
+## Using the ModelSerializer
+
+The default serialization options can be overridden by passing in a `ModelSerializerOptions` object to the Serialize and Deserialize methods. Developers can set the format options to specify the serialization format such as XML, JSON, and BinaryData. Default `ModelSerializationOptions` use the `Wire` format and the `System.Text.Json` serializer. The following samples demonstrate how to use `ModelSerializer` class for `System.Text.Json` and `Newtonsoft.Json` serialization.
+
+## Using ModelSerializer for System.Text.Json
+In the following sample, we are using the default `ModelSerializerOptions`. This will allow the `ModelSerializer` to use the `System.Text.Json` serializer for all models. In the Deserialization sample, we are setting the `ModelSerializerFormat` in the Options to `Json`. This will serialize all properties including read-only and additional properties.
 
 ### Serialization
 
-```C# Snippet:ExplicitCast_Serialize
-PetStoreClient client = new PetStoreClient(new Uri("http://somewhere.com"), new MockCredential());
-DogListProperty dog = new DogListProperty("myPet");
-Response response = client.CreatePet("myPet", (RequestContent)dog);
-var response2 = client.CreatePet("myPet", RequestContent.Create(dog));
+```C# Snippet:SystemTextJson_Serialize
+DogListProperty dog = new DogListProperty
```

Comment:
Lets use a simpler model name like `Dog` or perhaps a real model from one of our services?  The ListProperty portion of this name is just to indicate a use case we are covering in our testing it doesn't mean anything for customers.

Reply:
Could we use one of the real models in the SDK? As currently written, it's really not clear how DogListProperty is designed, why it works with the model serializer, etc. If we want to stick with DogListProperty we need to show how it's defined.